api.php

```diff
@@ -42,9 +42,12 @@ function hash_equals($a, $b) {
 
 
 include('db_connector.php');
-	
-	if (!hash_equals($authKey, crypt($auth, $authKey))){
-		die (json_encode(array('type' => API_ERROR_403, 'content' => 'Authentication failed.')));
+
+	session_start();
+	if (! isset($_SESSION['user_logged']) || $_SESSION['user_logged'] != 1) {
+		if (!hash_equals($authKey, crypt($auth, $authKey))){
+			die (json_encode(array('type' => API_ERROR_403, 'content' => 'Authentication failed.')));
+		}
```

Comment:
do check for authentication only if the user isn't already logged on via the web front-end